### PR TITLE
Rename operator to che-workspace-operator & controller is che-workspace-controller

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Renames operator to che-workspace-operator & controller is che-workspace-controller

:tada: It also contains adding cache Go modules during compiling in docker that speed up local rebuilding a lot. I haven't move it in a separate PR since it would need a redundant work with a renaming controller.

:warning: Once it's merged the GitHub repository must be renamed to https://github.com/che-incubator/che-workspace-operator

### What issues does this PR fix or reference?
I expanded a bit scope of Revise DevWorkspace CR structure https://github.com/eclipse/che/issues/15783
But I believe it's a good thing to do as initial set up of Che Workspace Operator

### Is it tested? How?
Yeah! I run a newly built operator (`sleshchenko/che-workspace-controller:renamed`) and start Java MySQL sample successfully.

### CODE REVIEW

File: deploy/flatten/che-workspace-crd-operator.v0.0.1.clusterserviceversion-v0.0.1.yaml [hunk 1]
```diff
@@ -134,7 +134,7 @@ metadata:
         }
       ]
     capabilities: Basic Install
-  name: che-workspace-crd-operator.v0.0.1
+  name: che-workspace-controller.v0.0.1
```
Comment (davidfestal) [inline deploy/flatten/che-workspace-crd-operator.v0.0.1.clusterserviceversion-v0.0.1.yaml]: Don't you also rename the file ?

File: deploy/flatten/che-workspace-crd.package.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 packageName: che-workspace-crd
 channels:
 - name: alpha
-  currentCSV: che-workspace-crd-operator.v0.0.1
```
Comment (davidfestal) [inline deploy/flatten/che-workspace-crd.package.yaml]: Don't you also rename the file ?

File: deploy/flatten/che-workspace-crd-operator.v0.0.1.clusterserviceversion-v0.0.1.yaml (hunk 1 above)
Comment (sleshchenko) [reply deploy/flatten/che-workspace-crd-operator.v0.0.1.clusterserviceversion-v0.0.1.yaml]: @davidfestal Could you suggest what is the right name for file and this field?
che-workspace-operator or che-workspace-controller

I guess che-workspace-operator

File: deploy/flatten/che-workspace-crd.package.yaml (hunk 2 above)
Comment (sleshchenko) [reply deploy/flatten/che-workspace-crd.package.yaml]: The file is renamed to `che-workspace-operator.package.yaml`

File: deploy/flatten/che-workspace-crd-operator.v0.0.1.clusterserviceversion-v0.0.1.yaml (hunk 1 above)
Comment (sleshchenko) [reply deploy/flatten/che-workspace-crd-operator.v0.0.1.clusterserviceversion-v0.0.1.yaml]: Name is changed to 
`che-workspace-operator.v0.0.1`

File name now is `che-workspace-operator.v0.0.1.clusterserviceversion-v0.0.1.yaml`

File: deploy/olm-catalog/che-workspace-crd/0.0.1/che-workspace-crd-operator.v0.0.1.clusterserviceversion.yaml [hunk 3]
```diff
@@ -134,7 +134,7 @@ metadata:
         }
       ]
     capabilities: Basic Install
-  name: che-workspace-crd-operator.v0.0.1
+  name: che-workspace-controller.v0.0.1
```
Comment (amisevsk) [inline deploy/olm-catalog/che-workspace-crd/0.0.1/che-workspace-crd-operator.v0.0.1.clusterserviceversion.yaml]: Does this file also need to be renamed?

File: deploy/olm-catalog/che-workspace-crd/che-workspace-crd.package.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
 packageName: che-workspace-crd
 channels:
 - name: alpha
-  currentCSV: che-workspace-crd-operator.v0.0.1
\ No newline at end of file
+  currentCSV: che-workspace-operator.v0.0.1
```
Comment (amisevsk) [inline deploy/olm-catalog/che-workspace-crd/che-workspace-crd.package.yaml]: Trailing newline

File: deploy/olm-catalog/che-workspace-crd/0.0.1/che-workspace-crd-operator.v0.0.1.clusterserviceversion.yaml (hunk 3 above)
Comment (sleshchenko) [reply deploy/olm-catalog/che-workspace-crd/0.0.1/che-workspace-crd-operator.v0.0.1.clusterserviceversion.yaml]: This file is renamed to `che-workspace-controller...`
`che-workspace-crd` olm package renamed to `che-workspace-operator`

**Pull request merged**